it_hardware/demo/purchase_order_confirm.xml

```diff
@@ -74,4 +74,40 @@
         <value model="stock.picking" eval="obj().env.ref('it_hardware.purchase_order_3').picking_ids.ids"/>
     </function>
     <function model="purchase.order" name="action_create_invoice" eval="[ref('purchase_order_3')]"/>
+
+    <function model="purchase.order" name="button_confirm" eval="[ref('purchase_order_4')]"/>
+    <function name="write" model="stock.move.line">
+        <value model="stock.move.line" eval="obj().search([
+        ('product_id', '=', ref('product_product_9')),
+        ('picking_id', '=', obj().env.ref('it_hardware.purchase_order_4').picking_ids.id)])[0].id"/>
+        <value model="stock.lot" eval="{'lot_id': ref('stock_lot_11'), 'lot_name': obj().env.ref('it_hardware.stock_lot_11').name}"/>
```

Comment:
(nitpicking) is it mandatory to provide the lot_name? Isn't it automatically retrieved by the lot_id.name?